backend/setup_env.py

```diff
@@ -0,0 +1,174 @@
+"""
+Script built on dotenv to set up the required environment variables
+"""
+
+import argparse
+from os import write
+from pathlib import Path
+from typing import Optional
+
+import dotenv
+
+# Assumes this file is in `/backend/setup_env.py`
+PROJECT_ROOT: Path = Path(__file__).parent.parent
+ENV_DIR: Path = PROJECT_ROOT.joinpath('env')
+
+BACKEND_ENV: Path = ENV_DIR.joinpath("backend.env")
+FRONTEND_ENV: Path = ENV_DIR.joinpath("frontend.env")
+MONGO_ENV: Path = ENV_DIR.joinpath('mongodb.env')
+SESSIONSDB_ENV: Path = ENV_DIR.joinpath('sessionsdb.env')
+    
+
+def main() -> None:
+    if not ENV_DIR.exists():
+        ENV_DIR.mkdir()
+
+    env = EnvReader(env_files=[BACKEND_ENV, FRONTEND_ENV, SESSIONSDB_ENV])
+
+    backend_env: dict[str, str] = {}
+    frontend_env: dict[str, str] = {}
+    mongo_env: dict[str, str] = {}
+    sessionsdb_env: dict[str, str] = {}
+
+    # sessionsdb - backend
+    sessionsdb_username = env.get_variable('SESSIONSDB_USERNAME', "name")
+    sessionsdb_pass = env.get_variable('SESSIONSDB_PASSWORD', "pass123")
+    sessionsdb_hostname = env.get_variable("SESSIONSDB_SERVICE_HOSTNAME", "sessiondb")
+
+    # cse auth - backend
+    auth_cse_client_id = env.get_variable("AUTH_CSE_CLIENT_ID", "...")
+    auth_cse_client_secret = env.get_variable("AUTH_CSE_CLIENT_SECRET", "...")
```

Comment:
thoughts on making these default to `None`? and then handle it by filtering out `None` env vars later or dont add them or whatever? (it is much conceptually and programmatically nicer for me to handle these as missing env vars then a empty flag string)

Also if you do this, mayb do invariant of either all or none of the "auth group" get written. (only if it doesnt make this ugly or cause big refactor, i make sure all or nothing is defined anyway when reading them again)

Reply:
oo sick

Reply:
wait i dont understand though, so if they don't provide auth credentials, it doesnt write entire backend.env?

Reply:
OH, auth goes in `backend.env`, im silly :sob: 

I forgot that auth wasn't its own file

Reply:
Okay so one solution here is to take in a list of `require_together` sets where for the elements in that set, if there is anything missing from one of those sets, it removes all of the other elements in that set.

We could also just hardcode this.

However, if it's not a big deal, I think we can just remove this check, since we are checking when we read these that everything is required anyways